htdocs/js/ui_utils.js

```diff
@@ -182,6 +183,41 @@ ui_utils.install_common_ace_key_bindings = function(widget, get_language) {
                         shell.scroll_to_end();
                     });
             }
+        },
+
+        {
+            name: 'execute-cursor-at-beginning',
+            bindKey: {
+          
+                mac: 'Ctrl-A',
+                sender: 'editor'
+            },
+            exec: function(widget, args, request) {
+                if (widget.getOption("readOnly"))
+                    return;
+                //row of the cursor on current line
+                var row = widget.getCursorPosition().row;
+                //move to the beginning of that line
+                widget.navigateTo(row, 0);
+                //make sure it appears at beginning of text
+                widget.navigateLineStart();
```

Comment:
yep, i was actually about to change those names. I'll repush that.

Without navigateLineStart, the cursor will go to direct beginning of the
line, so if you have any blank space before code, it will be placed before
the blank space instead of before the actual text. widget.navigateLineStart();
ensures the cursor is placed before actual text and not the empty space on
the line. Do you know if there's a better way to do that?

On Thu, Apr 23, 2015 at 2:29 PM, Gordon Woodhull <notifications@github.com>
wrote:

> In htdocs/js/ui_utils.js
> <https://github.com/att/rcloud/pull/1417#discussion_r28989783>:
>
> > +        {
> > +            name: 'execute-cursor-at-beginning',
> > +            bindKey: {
> > +
> > +                mac: 'Ctrl-A',
> > +                sender: 'editor'
> > +            },
> > +            exec: function(widget, args, request) {
> > +                if (widget.getOption("readOnly"))
> > +                    return;
> > +                //row of the cursor on current line
> > +                var row = widget.getCursorPosition().row;
> > +                //move to the beginning of that line
> > +                widget.navigateTo(row, 0);
> > +                //make sure it appears at beginning of text
> > +                widget.navigateLineStart();
>
> out of curiosity, what happens without navigateLineStart? it seems like
> it should be redundant.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/att/rcloud/pull/1417/files#r28989783>.
>



-- 
Anatoliy